examples/java-matter-controller/java/src/com/matter/controller/commands/common/MatterCommand.java

```diff
@@ -100,39 +103,22 @@ public void run() throws Exception {
 
   protected abstract void runCommand();
 
-  public void setTestResult(String result) {
-    mTestResult = Optional.of(result);
+  public void setSuccess() {
+    RealResult realResult = new RealResult();
+    realResult.setSuccess();
+    mFutureResult.setRealResult(realResult);
   }
 
-  public void expectSuccess(long timeout) {
-    expectResult("Success", timeout);
+  public void setFailure(String error) {
+    RealResult realResult = new RealResult();
+    realResult.setFailure(error);
+    mFutureResult.setRealResult(realResult);
   }
 
-  private void expectResult(String expectedResult, long timeout) {
-    long start = System.currentTimeMillis();
-    while (!mTestResult.isPresent())
-      try {
-        if (System.currentTimeMillis() > (start + timeout)) {
-          throw new RuntimeException("timeout!");
-        }
-        Thread.sleep(100);
-      } catch (InterruptedException ex) {
-      }
-
-    if (!mTestResult.isPresent()) {
-      throw new RuntimeException("received empty test result");
-    }
-
-    if (!mTestResult.get().equals(expectedResult)) {
-      if (!expectedResult.equals("Success")) {
-        System.out.format(
-            "%s command failed:%n    Expected: %s%n    Got: %s%n",
-            getName(), expectedResult, mTestResult);
-        throw new RuntimeException(getName());
-      } else {
-        System.out.format("%s command failed: %s%n", getName(), mTestResult.get());
-      }
+  public void waitComplete(long timeout) {
+    mFutureResult.setTimeout(timeout);
+    if (mFutureResult.getResult()) {
+      logger.log(Level.INFO, "Expect result received");
```

Comment:
Do we need this log? Feels debug-like. If we do need it, maybe we should log the result too.

Reply:
removed, thanks